### ISSUE TITLE
Consider using directives over magic comments

### ISSUE THREAD

**Issue by LegNeato** [davidpdrsn/juniper-from-schema#27]
GraphQL has [directives](https://graphql.github.io/graphql-spec/draft/#sec-Language.Directives) which are used for telling the client / server additional information about desired implementation. It might be better and fit other tooling expectations to use directives rather than comments for declaring a field owned.

**Comment by davidpdrsn** [comment]
Thanks for the tip 😄 I also wasn't a fan of the magic comments. Honestly haven't looked into directives but I'll read up on them.

**Comment by davidpdrsn** [comment]
@LegNeato I've looked into schema directives and it seems like a much nicer solution than the magic comments. Just to be sure, before I start coding, is it something like this you had in mind:

```graphql
directive @owned on FIELD_DEFINITION

type Query {
    string: String! @owned
}

schema { query: Query }
```

Instead of

```graphql
type Query {
    "#[ownership(owned)]"
    string: String!
}

schema { query: Query }
```

**Comment by LegNeato** [comment]
Yep! Or, if you want you could do something like `@rust(ownership: owned)` or `@juniper(ownership: owned)`.

I don't think it makes too much difference and is more your preference.

**Comment by LegNeato** [comment]
Then again, might be worth standardizing on semantics in the rust graphql ecosystem so everything can read and write and keep the same rust-isms.

**Comment by davidpdrsn** [comment]
Yeah thats probably a good idea. How do you suggest we do that? Should we bring in more people here or should I open an issue on the juniper repo?
